x-pack/plugins/ml/server/routes/trained_models.ts

```diff
@@ -54,6 +57,45 @@ export function filterForEnabledFeatureModels<
   return filteredModels;
 }
 
+export const getInferenceServicesProvider = (client: IScopedClusterClient) => {
+  return async function getInferenceServices(
+    trainedModels: TrainedModelConfigResponse[],
+    asInternal: boolean = false
+  ) {
+    const esClient = asInternal ? client.asInternalUser : client.asCurrentUser;
+
+    try {
+      // Check if model is used by an inference service
+      const { models } = await esClient.transport.request<{
+        models: InferenceAPIConfigResponse[];
+      }>({
+        method: 'GET',
+        path: `/_inference/_all`,
+      });
+
+      const inferenceAPIMap = groupBy(
+        models,
+        (model) => model.service === 'elser' && model.service_settings.model_id
+      );
+
+      for (const model of trainedModels) {
+        if (model.model_id in inferenceAPIMap) {
+          if (!asInternal) {
+            model.inference_apis = inferenceAPIMap[model.model_id];
+          }
+          model.hasInferenceServices = !!inferenceAPIMap[model.model_id];
```

Comment:
This will always be true. It needs to be moved out of the `if (model.model_id in inferenceAPIMap) {` block.

It might be more efficient to do something like this, rather than accessing the map three times.
```ts
      for (const model of trainedModels) {
        const inferenceApis = inferenceAPIMap[model.model_id];
        model.hasInferenceServices = !!inferenceApis;
        if (model.hasInferenceServices && !asInternal) {
          model.inference_apis = inferenceApis;
        }
      }
```

Reply:
Refactored in https://github.com/elastic/kibana/pull/175880/commits/5ac0bfd18dc70f38b42a91b510478cb2ef6bb433